src/components/PopupWithForm.js

```diff
@@ -18,17 +19,40 @@ export default class PopupWithForm extends Popup {
     return formValues;
   }
 
+  _setInputValues(data) {
+    this._inputList.forEach((input) => {
+      input.value = data[input.name] || "";
+    });
+  }
+
   _setEventListeners() {
-    this._popupForm.addEventListener("submit", (event) => {
+    this._popupForm.addEventListener("submit", async (event) => {
       event.preventDefault();
       const data = this._getInputValues();
-      this._handleFormSubmission(data);
-      this.reset();
+      console.log("Form data:", data);
+      this.renderLoading(true);
+      try {
+        await this._handleFormSubmission(data);
+        setTimeout(() => {
+          this.close();
```

Comment:
`setTimeout` is not needed when you submit any form because you need to really **wait** for the server response

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well